packages/nimble-components/build/generate-icons/source/index.js

```diff
@@ -70,10 +90,18 @@ export const ${tagName} = DesignSystem.tagFor(${className});
     fileCount += 1;
 
     allIconsFileContents = allIconsFileContents.concat(`export { ${className} } from './${fileName}';\n`);
+    mdxFileContents = mdxFileContents.concat(`${mdxImportStatements}\n`);
+    mdxTableFileContents = mdxTableFileContents.concat(`${mdxIconTableRow}\n`);
 }
 console.log(`Finshed writing ${fileCount} icon component files`);
```

Comment:
I think it would be easier to follow the output document structure if we kept all the MDX stuff together inside the loop and concatenated the parts of the output file all at once after the loop. So something more like this:
```ts
let mdxImportStatements = '';
let mdxTable = mdxIconTablePrefix;
for (const key of Object.keys(icons)) {
    // ...
    mdxImportStatements += `import { ${tagName} } from './${fileName}';\n`;
    mdxTable += `| <IconItem name="${fileName}"><${elementName} /></IconItem> | ${iconSynonyms?.tags.join(', ')} |\n`;
    // ...
}

const mdxFileContents = `
${mdxFilePrefix}
${mdxImportStatements}
${mdxTable}`;